.github/workflows/build-ios.yml

```diff
@@ -0,0 +1,63 @@
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}-build-ios"
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
+
+jobs:
+  build-ios:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        task:
+          - ios-device
+          - ios-simulator-x86
+          - ios-simulator-arm
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: determine rust target
+        id: rust-target
+        run: |
+          if [[ "${{ matrix.task }}" == "ios-device" ]]; then
+            echo "target=aarch64-apple-ios" >> $GITHUB_ENV
+          elif [[ "${{ matrix.task }}" == "ios-simulator-x86" ]]; then
+            echo "target=x86_64-apple-ios" >> $GITHUB_ENV
+          elif [[ "${{ matrix.task }}" == "ios-simulator-arm" ]]; then
+            echo "target=aarch64-apple-ios-sim" >> $GITHUB_ENV
+          fi
+
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          target: ${{ env.target }}
+
+      - name: setup cargo-make
+        uses: davidB/rust-cargo-make@v1
+
+      - name: build ${{ matrix.task }}
+        run: |
+          cd crypto-ffi
+          cargo make ${{ matrix.task }}
+
+      - name: upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{github.event.number}}-${{ matrix.task }}
+          path: target
+          retention-days: 1
+          overwrite: 'true'
+      # Only needs to be uploaded once, this step finishes fastest.
+      - name: upload ffi artifact
+        if: startsWith(matrix.task, 'ios-simulator-arm')
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{github.event.number}}-swift-ffi
```

Comment:
Here we're using `github.event.number`, elsewhere we're using `github.run_id` to uniquely name artifacts. I think we should be consistent, and use `github.run_id` everywhere, unless there is a good reason to prefer `github.event.number` here.

Reply:
I just moved the job into separate file maybe @coriolinus can give a reason.

Reply:
I didn't touch that actually; it appears to come from https://github.com/wireapp/core-crypto/commit/1ce7d95b3e2267e08b30766644a58eadee349d30, which is @SimonThormeyer. 

I don't think there's a strong reason here to prefer either one here. TBH we'd probably be fine without a distinguishing number in either case; looking deeper into the download artifact action, it automatically partitions by run id anyway. Adding an event number or run id just helps us keep track of things if we happen to download some artifacts for manual inspection.

Reply:
> looking deeper into the download artifact action, it automatically partitions by run id anyway

Ah, didn't know that, thanks!